### PR TITLE
Integrate GPT4-vision support

### PR DESCRIPTION
**Pull request by srbalakr**
## Purpose

This PR adds multimodal capabilities to the RAG sample by leveraging azure open ai gpt4 -vision.
Please go through docs/gpt4.md to set up and experience the capability.

## Does this introduce a breaking change?

```
[x ] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ x] Feature
[ ] Code style update (formatting, local variables)
[x ] Refactoring (no functional changes, no api changes)
[x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -241,6 +241,24 @@ either you or they can follow these steps:
 
 ## Enabling optional features
 
+### Enabling GPT-4 Turbo with Vision
+
+This section covers the integration of GPT-4 Vision with Azure AI Search. Learn how to enhance your search capabilities with the power of image and text indexing, enabling advanced search functionalities over diverse document types. For a detailed guide on setup and usage, visit our [Enabling GPT-4 Turbo with Vision](docs/gpt4v.md) page.
+
+### Enabling Application Insights
```
Comment (pamelafox) [inline README.md:248]: This section should be removed (it exists below)

**Comment by pamelafox** [comment]
To get 3.9 CI happy, change this line in .github/workflows/python-test.yaml

run: python3 -m pytest -s -vv --cov --cov-fail-under=87

to =86

I didn't quite get coverage back up to 87% :(

File: infra/core/security/keyvault.bicep [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+param location string = resourceGroup().location
+param tags object = {}
+
+param principalId string = ''
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: name
+  location: location
+  tags: tags
+  properties: {
```
[the remark this answers is not in the document]
Comment (pamelafox) [reply infra/core/security/keyvault.bicep:12]: I think we just have to accept this for now :(

**Review by pamelafox — APPROVED** [review]
Looks as expected now. I can do another test azd up or you can merge now, depending when you need it live.

**Comment by srbalakr** [comment]
> Looks as expected now. I can do another test azd up or you can merge now, depending when you need it live.

I did a fresh azd up on Aust East and swiss nor. Looking good. Will merge it in sometime.

**Review by chuwik — APPROVED** [review]

**Comment by pamelafox** [comment]
Sigh am not sure why GitHub decided not to run the checks on the last commit, I had that problem earlier today with another repo. Tests now pass locally.

**Pull request merged**